api/agent_test.go

```diff
@@ -626,6 +626,60 @@ func TestAPI_AgentServiceAddress(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 }
+func TestAPI_AgentServiceSocket(t *testing.T) {
+	t.Parallel()
+	c, s := makeClient(t)
+	defer s.Stop()
+
+	agent := c.Agent()
+
+	reg1 := &AgentServiceRegistration{
+		Name:    "foo1",
+		Port:    8000,
+		Address: "192.168.0.42",
+	}
+	reg2 := &AgentServiceRegistration{
+		Name:       "foo2",
+		SocketPath: "/tmp/foo2.sock",
+	}
+
+	if err := agent.ServiceRegister(reg1); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if err := agent.ServiceRegister(reg2); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	services, err := agent.Services()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if _, ok := services["foo1"]; !ok {
+		t.Fatalf("missing service: %v", services)
+	}
+	if _, ok := services["foo2"]; !ok {
+		t.Fatalf("missing service: %v", services)
```

Comment:
Can you convert the assertions in this test to use the `require` package? We've been using it for all new test assertions, like:
- `require.NoError(t, err)`
- `require.Contains(t, services, "foo1")`
- `require.Equal(t, "192.168.0.42", services["foo1"].Address)`
- `require.Empty(t, services["foo2"].Address)`

etc

For example, currently if this assertion failed the message would be:
`missing service: map[foo1:0xc0001bfb00]`, which doesn't have the context of the key we were looking for.